src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicWorkspacePullDiagnosticsHandler.cs

```diff
@@ -118,4 +137,41 @@ protected override WorkspaceDiagnosticPartialReport CreateUnchangedReport(TextDo
             }
         }).ToImmutableArray();
     }
+
+    private void OnLspSolutionChanged(object? sender, WorkspaceChangeEventArgs e)
+    {
+        UpdateLspChanged();
+    }
+
+    private void OnLspTextChanged(object? sender, EventArgs e)
+    {
+        UpdateLspChanged();
+    }
+
+    private void UpdateLspChanged()
+    {
+        Interlocked.Exchange(ref _lspChanged, 1);
+    }
+
+    protected override async Task WaitForChangesAsync(RequestContext context)
```

Comment:
💡 Consider adding another way to exit this (some sort of shutdown cancellation token)

Reply:
🤦 ... yes this should absolutely use the cancellation token passed in by the queue.  that'll take care of either request cancellation / shutdown.  fixed!